client/js/components/day/day.js

```diff
@@ -12,32 +16,41 @@ var Day = React.createClass({
     // pending: array full of objects representing employees
     return {
       date: '',
-      gigs: [],
-      pending: []
+      gigs: {}
     };
   },
 
   componentWillMount: function(){
-    DayStore.addChangeListener(this._handleUpdate);
-     // trigger API call to get initial data.
-    ViewActionCreator.getDayData();
+    DayStore.addChangeListener(this._handleDayStoreUpdate);
+    EmployeeStore.addChangeListener(this._handleEmployeeStoreUpdate);
+
+    var today = moment();
+
+    ViewActionCreator.getAvailableEmployees(today);
+    ViewActionCreator.getDayData(today);
   },
 
   componentWillUnmount: function(){
-    DayStore.removeChangeListener(this._handleUpdate);
+    EmployeeStore.removeChangeListener(this._handleEmployeeStoreUpdate);
+    DayStore.removeChangeListener(this._handleDayStoreUpdate);
   },
 
-  _handleUpdate: function(){
+  _handleDayStoreUpdate: function(){
     this.setState(DayStore.getData());
   },
 
+  _handleEmployeeStoreUpdate: function(){
+    this.render();
+  },
+
   render: function(){
-    var gigs = this.state.gigs.map(function(gig, idx){
+    var gigs = _.map(this.state.gigs, function(gig, idx){
       return (
         <div className='bin day-bin'>
           <GigBin
-            information={gig.information}
-            staff={gig.staff}
+            information={gig}
```

Comment:
I'm confused about what you mean by duplicating information - we are just passing a reference to the gig object, not making a copy. There will only be one object in memory. I agree that the information bucket has access to extra information (gig.Users and gig.Positions) but I don't think it is worth creating a new property on gig which contains just the info - that'd involve more data manipulation either on the server or on the client. Can you explain a good reason not to do this?